### PR TITLE
Don't create wxWindowAccessible by default

### PR DESCRIPTION
**Pull request by vslavik**
**tl;dr:** This reduces the impact of `wxUSE_ACCESSIBILITY` to only windows that actively use it, instead of the current all-or-nothing situation that degrades accessibility interface of all windows. It also makes it safe to enable the flag by default (I think we should, especially in light of Artur's work).

The change deserves a longer explanation. Let me preface it by saying that I _really_ didn’t want to do this. I’ve been agonizing over it for a week, but ultimately didn’t find any other acceptable solution.

The problem it addresses is three-fold:
1. `wxUSE_ACCESSIBILITY` is not in a good shape. The last non-trivial change to its `IAccessible` code was in 2009(!), the last _substantial_ change is even older. `IAccessible` isn’t fully implemented, neither is `wxWindowAccessible`, most of which is doing things native code already does, sometimes in worse ways (focus problems, not getting proper labels for anything but wxButtons).
2. It’s off by default, making it very unlikely for bugs to be fixed or even noticed.
3. It is _intrusive_. Enable the flag and suddenly your application starts behaving very differently under screen readers or MSAA inspectors. It replaces perfectly good standard Windows implementations of `IAccessible` for common controls or even for things like a plain container window with wx's implementation. There’s nothing gained by doing that and due to 1., it actually _degrades_ the quality of accessibility support by substituting inferior implementations.

Let me show you an example of Poedit inspected with `inspect.exe`. This is wx with no accessibility whatsoever:

accessibility_off

And this is with `wxUSE_ACCESSIBILITY` enabled:

vanilla_wx

The latter is worse in highlighted areas and that’s without considering the behavior.

So what this patch does (finally!) is that it reduces the _impact_ of accessibility code by making sure it is used only when it servers a useful role: in other words, for custom controls with custom `wxAccessible` implementations. `wxWindowAcessible` is still useful as their base class, but it is no longer used as the default implementation for anything.

This has compatibility consequences:
1. Code may be expecting accessible objects always, no longer true.
2. We loose the “help” text hooks into help providers. I looked around in various applications and didn’t see this property set anywhere. If needed, explicitly instantiating `wxWindowAccessible` fixes it.

But most importantly, it makes wx apps behave as expected even with `wxUSE_ACCESSIBILITY` enabled, while still allowing custom controls to do the right thing.

Ideally, this would be followed by more work: namely removing all the redundant implementations from `wxWindowAccessible` and removing any extra work from `wxIAccessible` so that it servers as a pure proxy to the standard implementation, allowing custom controls to easily customize only the relevant bits of functionality. But the most important part is taking `wxIAccessible` out of the equation in 99% of places where no customization is intended.

Some other alternatives I considered:
1. Just fix the wx implementation. Unrealistic and a moving target.
2. Cut the wx code to the bone, only leave things that are not-default (help text) and rigorously proxy everything else to the native standard impl. from `CreateStdAccessibleObject` (or `CreateStdAccessibleProxy`, but I don’t think wx is subclassing in a way that matters here, is it?). Nice in theory, but the standard proxy _also_ doesn’t behave as the native thing returned from `WM_GETOBJECT`. Probably because it is, per MSDN, intended to help with implementing custom controls, not to be used for _every single window_.
3. Likewise, but use the object returned from default wnd proc `WM_GETOBJECT`. This would require use of `ObjectFromLresult`, which is explicitly documented as being for internal use and “should not call”. I’m not comfortable with using it.
4. Have `wxUSE_ACCESSIBILITY_SUPPORT` in addition to the old setting, which would differ from `wxUSE_ACCESSIBILITY` only in lack of `CreateAccessible` implementation — it would provide the API needed by custom controls, but wouldn’t force the impl. everywhere. But I think the behavior pre-this-patch was so buggy that it doesn’t make sense to retain it.

It’s also worth noting that Microsoft realized MSAA was deeply flawed, hence UI Automation, and _even within MSAA_, providing custom `IAccessible` is meant for the custom control scenario that needs it. Simple accessibility annotations (the kind that any wx app would benefit from) can and should be done in a lighter way with properties annotation. So it really makes little sense to force the custom impl in such a heavy-handed way.

So in the end, I think this small change, even though not backwards compatible, really is the best thing to do.

/cc @vadz @a-wi

### CODE REVIEW

**Comment by vadz** [comment]
In principle, we could avoid formally breaking compatibility by continuing to return `new wxWindowAccessible` from `wxWindowBase::CreateAccessible()` but setting some "not really needed" flag on it there, so that the callers, e.g. `WM_GETOBJECT` handler in `src/msw/window.cpp`, could check for it and not handle the message in this case.

This being said, I think it's not worth it and your solution is fine for the reasons you've explained so well, so I'm OK with just doing this instead.

Thanks!
